File: android/app/src/main/java/com/mitoast/model/NotificationMessage.kt
package com.mitoast.model

import kotlinx.serialization.Serializable

@Serializable
data class NotificationMessage(
    val type: String = "notification",
    val id: String,
    val key: String,
    val packageName: String,
    val appName: String,
    val title: String,
    val content: String,
    val bigTitle: String = "",
    val bigText: String = "",
    val subText: String = "",
    val ticker: String = "",
    val hintTitle: String = "",
    val hintText: String = "",
    val iconBase64: String = "",
    val timestamp: Long,
    val category: String = "general",
    val isOngoing: Boolean = false,
    val groupKey: String = "",
    val peopleCount: Int = 0,
    val progress: NotificationProgress? = null,
    val mediaActions: List<MediaAction>? = null,
    /** 媒体是否正在播放（来自 MediaSession PlaybackState） */
    val mediaIsPlaying: Boolean = false,
    /** 媒体当前播放位置（毫秒），-1 表示未知 */
    val mediaPositionMs: Long = -1,
    /** 媒体总时长（毫秒），-1 表示未知 */
    val mediaDurationMs: Long = -1
)

@Serializable
data class NotificationProgress(
    val current: Int,
    val total: Int,
    val label: String
)

@Serializable
data class MediaAction(
    val name: String,
    val index: Int,
    /** 通知 action 的原始文案（如"已收藏/取消收藏/关闭歌词"），用于推断开关初始态 */
    val title: String = ""
)

/** 妙播/投放在内的播放设备（本机、蓝牙、Cast 设备、智能音箱等） */
@Serializable
data class CastDevice(
    val id: String,
    val name: String,
    /** phone / tv / bt / speaker */
    val deviceType: String,
    val selected: Boolean = false
)

@Serializable
data class CastDevicesMessage(
    val type: String = "cast_devices",
    val devices: List<CastDevice> = emptyList(),
    /**
     * 当前为 HyperOS 小米妙播生态（系统内置 com.milink.service）。
     * 小爱音箱/小米电视/小米电脑等 Wi-Fi 妙播设备不走标准 MediaRouter2，第三方无法枚举/直连，
     * 需在手机控制中心「小米妙播」面板选择，PC 端仅做提示与状态展示。
     */
    val miplaySupported: Boolean = false,
    /** 音频是否正通过小米妙播投放到音箱/电视/电脑（Settings.Global: miplay_audio_cast_state）。 */
    val miplayCasting: Boolean = false,
    /** 音频是否正妙播到车机（Settings.Global: ucar_casting_state）。 */
    val carCasting: Boolean = false,
    /** 妙播可由 PC 端反向控制（无障碍服务可用，可自动完成选择器操作）。 */
    val miplayControllable: Boolean = false
)

@Serializable
data class PingMessage(
    val type: String = "ping",
    val deviceName: String,
    val deviceModel: String,
    val version: String = "1.0.0"
)

@Serializable
data class ClearNotificationMessage(
    val type: String = "clear",
    val key: String
)

File: windows/src/MiToast/Models/NotificationMessage.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace MiToast.Models;

public class NotificationProgress
{
    [JsonPropertyName("current")]
    public int Current { get; set; }

    [JsonPropertyName("total")]
    public int Total { get; set; }

    [JsonPropertyName("label")]
    public string Label { get; set; } = string.Empty;
}

public class MediaAction
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("index")]
    public int Index { get; set; }

    /// <summary>原始 action 文案（用于推断收藏/歌词开关当前状态）</summary>
    [JsonPropertyName("title")]
    public string Title { get; set; } = string.Empty;
}

public class NotificationMessage
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "notification";

    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    [JsonPropertyName("key")]
    public string Key { get; set; } = string.Empty;

    [JsonPropertyName("packageName")]
    public string PackageName { get; set; } = string.Empty;

    [JsonPropertyName("appName")]
    public string AppName { get; set; } = string.Empty;

    [JsonPropertyName("title")]
    public string Title { get; set; } = string.Empty;

    [JsonPropertyName("content")]
    public string Content { get; set; } = string.Empty;

    [JsonPropertyName("bigTitle")]
    public string BigTitle { get; set; } = string.Empty;

    [JsonPropertyName("bigText")]
    public string BigText { get; set; } = string.Empty;

    [JsonPropertyName("subText")]
    public string SubText { get; set; } = string.Empty;

    [JsonPropertyName("ticker")]
    public string Ticker { get; set; } = string.Empty;

    [JsonPropertyName("hintTitle")]
    public string HintTitle { get; set; } = string.Empty;

    [JsonPropertyName("hintText")]
    public string HintText { get; set; } = string.Empty;

    [JsonPropertyName("iconBase64")]
    public string IconBase64 { get; set; } = string.Empty;

    [JsonPropertyName("timestamp")]
    public long Timestamp { get; set; }

    [JsonPropertyName("category")]
    public string Category { get; set; } = "general";

    [JsonPropertyName("isOngoing")]
    public bool IsOngoing { get; set; }

    [JsonPropertyName("groupKey")]
    public string GroupKey { get; set; } = string.Empty;

    [JsonPropertyName("peopleCount")]
    public int PeopleCount { get; set; }

    [JsonPropertyName("progress")]
    public NotificationProgress? Progress { get; set; }

    [JsonPropertyName("mediaActions")]
    public List<MediaAction>? MediaActions { get; set; }

    [JsonPropertyName("mediaIsPlaying")]
    public bool MediaIsPlaying { get; set; }

    [JsonPropertyName("mediaPositionMs")]
    public long MediaPositionMs { get; set; } = -1;

    [JsonPropertyName("mediaDurationMs")]
    public long MediaDurationMs { get; set; } = -1;
}

/// <summary>妙播/投放在内的播放设备（本机、蓝牙、Cast 设备等）</summary>
public class CastDevice
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    /// <summary>设备类型图标：phone（本机）/ bt（蓝牙）/ tv（投屏）/ speaker（音箱）</summary>
    [JsonPropertyName("deviceType")]
    public string DeviceType { get; set; } = "speaker";

    [JsonPropertyName("selected")]
    public bool Selected { get; set; }
}

public class CastDevicesMessage
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "cast_devices";

    [JsonPropertyName("devices")]
    public List<CastDevice> Devices { get; set; } = new();

    /// <summary>HyperOS 小米妙播生态（小爱音箱/小米电视等 Wi-Fi 设备不在 devices 列表中，需在手机控制中心选择）。</summary>
    [JsonPropertyName("miplaySupported")]
    public bool MiplaySupported { get; set; }

    /// <summary>音频正通过小米妙播投放到音箱/电视/电脑。</summary>
    [JsonPropertyName("miplayCasting")]
    public bool MiplayCasting { get; set; }

    /// <summary>音频正妙播到车机。</summary>
    [JsonPropertyName("carCasting")]
    public bool CarCasting { get; set; }

    /// <summary>妙播可由 PC 反向控制（手机端无障碍服务可用，可自动操作系统选择器）。</summary>
    [JsonPropertyName("miplayControllable")]
    public bool MiplayControllable { get; set; }
}

public class ClearNotificationMessage
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "clear";

    [JsonPropertyName("key")]
    public string Key { get; set; } = string.Empty;
}

public class PingMessage
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "ping";

    [JsonPropertyName("deviceName")]
    public string DeviceName { get; set; } = string.Empty;

    [JsonPropertyName("deviceModel")]
    public string DeviceModel { get; set; } = string.Empty;

    [JsonPropertyName("version")]
    public string Version { get; set; } = "1.0.0";
}
